app/src/androidTest/java/com/lightricks/feedexercise/network/MockFeedApiService.kt

```diff
@@ -1,5 +1,37 @@
 package com.lightricks.feedexercise.network
 
-/**
- * todo: implement the mock feed API service here
- */
\ No newline at end of file
+import android.content.Context
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import io.reactivex.Single
+import okio.buffer
+import okio.source
+
+
+class MockFeedApiService(private val context: Context) : FeedApiService {
+
+    private var shouldThrowError = false
+
+    private val moshi = Moshi.Builder()
+        .addLast(KotlinJsonAdapterFactory())
+        .build()
+
+    override fun getFeed(): Single<GetFeedResponse> {
+        if (shouldThrowError) {
+            shouldThrowError = false
+            throw MockNetworkException()
```

Comment:
Usually an error happens during the network request. 
You can simulate this by returning `Single.error(MockNetworkException())`.